api/lib/source/cosmosV0-source.js

```diff
@@ -224,14 +227,113 @@ class CosmosV0API extends RESTDataSource {
       validator.signing_info = signingInfos[consensusAddress]
     })
 
-    return validators.map((validator) =>
+    const validatorsWithoutProfiles = validators.map((validator) =>
       this.reducers.validatorReducer(
         this.network.id,
         signedBlocksWindow,
         validator,
         annualProvision
       )
     )
+    if (profile) {
+      return { validators, validatorsWithoutProfiles }
+    } else {
+      return validatorsWithoutProfiles
+    }
+  }
+
+  async getProfilesForValidators(
+    primitiveValidators,
+    validatorsWithoutProfiles,
+    fiatCurrency = 'USD'
+  ) {
+    // const networkList = await db.getNetworks()
+    validatorsWithoutProfiles = this.getRanksForValidators(
+      validatorsWithoutProfiles
+    )
+    return await Promise.all(
+      validatorsWithoutProfiles.map(async (validator) => {
+        const [
+          validatorProfile,
+          // latestValidatorNotifications,
+          allValidatorDelegations
+        ] = await Promise.all([
+          this.db.getValidatorProfile(validator.operatorAddress),
+          // this.db.getAccountNotifications(
```

Comment:
delete?

Reply:
so I understand you don't agree with having the query taking that long, right? It works, like I said, but is is very slow. That's all (like Polkadot in ancient times)

Reply:
i mean it's commented out so just remove it from the PR

Reply:
the question if we should do this (uncomment) or not (remove)